Bottom: f1461e27a310bcba5bc27c10f61f02e952bdffd2
Top:    5443b430419fc07b750ee98358ff646fef9e996a
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-16 17:41:03 +0800

Refresh of xdp-introduce-xdp-free-func

---

diff --git a/include/linux/filter.h b/include/linux/filter.h
index 13fdcd6..6c50b5a 100644
--- a/include/linux/filter.h
+++ b/include/linux/filter.h
@@ -438,9 +438,14 @@ struct xdp_buff {
 	void *ring;
 	void *private;
 	struct net_device *netdev;
-	void (*free) (const void *ring, const void *private);
+	int (*free) (const void *ring, const void *private);
 };
 
+static inline int xdp_buff_free(struct xdp_buff *xdp_buff)
+{
+	return xdp_buff->free(xdp_buff->ring, xdp_buff->private);
+}
+
 static inline void xdp_buff_init(struct xdp_buff *xdp_buff,
                                  void *data, void *data_end,
                                  void *ring, void *private,
